core/src/main/java/org/mskcc/cbio/portal/dao/DaoMutation.java

```diff
@@ -1564,4 +1564,37 @@ public static boolean hasDriverAnnotations(String _cancerStudyStableIds) throws
         }
         return false;
     }
+
+    /**
+     * Returns true if there are duplicates in `mutation_event`.
+     * Compares count(*) of records in `mutation_event` against the count(distinct ...)
+     * records based on the fields specified below.
+     *
+     * If the counts do not match then SQL statement returns 0 (false), otherwise returns 1 (true).
+     * @return
+     * @throws DaoException
+     */
+    public static boolean hasDuplicateMutationEvents() throws DaoException {
+        Connection con = null;
+        PreparedStatement pstmt = null;
+        ResultSet rs = null;
+
+        Integer returnValue = -1; // default = -1 to indicate mismatched counts from db table
```

Comment:
about the comment - if we reach the end of the function and the value is still -1, it would mean that there was an exception during the running of the query (the query did not complete) and we will not know whether duplicate records exist or not. If the query does complete, this value will be either 0 (duplicate records exist) or 1 (all records are distinct).

Also, I think the name 'returnValue' is commonly used to hold a value which will be returned when the function exits. This variable holds a value returned from the SQL resultSet, so maybe a clearer name would be something like "DistinctRecordQueryResult". Alternatively, we *could* end this function with `return returnValue` .. so this function would return one of three (Integer) values : -1 (don't know), 0 (duplicated records exist), 1 (all records are distinct)